docs/src/modules/java/pages/eventsourced.adoc

```diff
@@ -134,6 +134,53 @@ By default, the generated constructor has an `EventSourcedEntityContext` paramet
 
 When more components are added the `AkkaServerlessFactory` is regenerated and you have to adjust the registration from the `Main` class.
 
+=== Unit Tests with autogenerated Testkit
+
+We can test this Event Sourced Entity implementation with the SchoppingCartTestkit. This class is autogenerated by AkkaServerless when this project is compiled.
+
+The following is a simplification of its contents but in essence, what we show in the next snippet is the signature of the available methods
+
+.target/generated-test-sources/com/example/shoppingcart/domain/ShoppingCartTestKit.java
+[source,java]
+----
+include::example$testkit/java-eventsourced-shopping-cart/intarget/generated-test-sources/akkaserverless/com/example/shoppingcart/domain/ShoppingCartTestKit.java[] 
+----
+<1> This constructor requires a `ShoppingCart` Entity so we can interact with this Entity through the `ShoppingCartTestKit`.
+<2> This constructor allows us to set a state on the Entity, so we can start the test directly from the desired state instead of having to send all the commands that would produce such a state.
+<3> This method returns the current state of the Entity. 
+<4> This method returns the accumulated events from all the executed commands sent to the entity.
+<5> This method calls to the method with the same name in the Event Source Entity passed in the constructor.
+
+
+Once we call an RPC method on the entity this gives us back a `Result`. This class has the following methods that we can use to assert our expectations.
+
+[source,java]
+----
+include::example$testkit/com/akkaserverless/javasdk/testkit/EventSourcedResult.java[tag=available-methods] 
```

Comment:
this could be a link to api docs instead

Reply:
And avoid the explanation of the callouts under this snippet altogether?

Reply:
let's leave it like this for now and follow up later